en/includes/guides/authentication/conditional-auth/on-demand-silent-password-migration-template.md

```diff
@@ -0,0 +1,378 @@
+    </tr>
+</table>
+
+## How it works
+
+Let's look at how the above conditional authentication scripts works.
+
+1. The first authentication step (Username & Password) is initiated with the `executeStep(1, ..)` function. Based on its status, one of the following happens.
+
+    - If the entered credentials match, the user's password is already migrated. Hence, the `onSuccess` function will be called and the user will be authenticated.
+    
+    - If the user's credentials don't match, either the credentials are incorrect or the password may not have be migrated yet. Hence, the `onFail` callback function is called and the script continues.
+
+2. If the `onFail` function is called, the script will next try to locate a unique user in the system using the `resolveAndInitUser` function. 
+
+    !!! note "Adjust script for alternate login identifiers"
+
+        If your organization does not use alternate login identifiers, uncomment the following lines in the `resolveAndInitUser` function.
+
+        ```js
+        Log.info("Resolving username using the login identifier: " + loginIdentifier);
+        username = resolveMultiAttributeLoginIdentifier(loginIdentifier, context.tenantDomain);
+        Log.info("Username resolved using the login identifier. Resolved username: " + username);
+        ```
+
+3. If a unique user is found, the script checks the value of the `is_migrated` attribute of the user and does one of the following. (The `is_migrated` user attribute holds the status of the password migration.)
+
+    - If this is set to `true`, user's password is already migrated. Hence, the entered credentials are incorrect and the flow fails with an error. 
+    
+    - If it is not set to `true`, the user's password is not yet migrated. Hence, the script calls for external authentication.
+
+4. The script calls for external authentication with the `authenticateExternally` function and it works as follows:
+
+    - The `connectionMetadata` object holds the start authentication endpoint URL, Choreo application credentials and the Asgardeo token endpoint.
+    - The script first calls the `callChoreo()` function along with the `connectionMetaData` and invokes the start authentication endpoint.
+    - If the API call is successful, the `onSuccess()` callback function is called which in turn calls the `prompt()` function.
+    - The `prompt()` function continuously polls the Choreo polling endpoint and redirects the user to a waiting page until the external authentication completes.
+    - Once the authentication is complete, the `onSuccess()` callback function of the `prompt()` function calls the `updatePasswordAndReAuthenticate()` function.
+
+5. The `updatePasswordAndReAuthenticate()` function is responsible for checking the status of the authentication and taking necessary actions as explained below.
+
+    - The `connectionMetadata` object hold the authentication status endpoint URL, Choreo application credentials and the Asgardeo token endpoint
```

Comment:
```suggestion
    - The `connectionMetadata` object hold the authentication status endpoint URL, Choreo application credentials and the Asgardeo token endpoint.
```

Reply:
Fixed with https://github.com/wso2/docs-is/commit/d0b7130af4b2f0857f356c6dbdcbce192e8484c2